src/object_api.c

```diff
@@ -42,6 +42,10 @@ git_repository *git_commit_owner(const git_commit *obj)
 	return git_object_owner((const git_object *)obj);
 }
 
+int git_commit_dup(git_commit **out, git_commit *obj)
+{
+	return git_object_dup((git_object **)out, (git_object *)obj);
```

Comment:
That's right, `git_object_dup` _is_ public.  I don't know why I avoided making the `git_commit_dup` and friends public.  They seem useful.